tests/acceptance/authorization-error-test.js

```
@@ -0,0 +1,48 @@
+import Ember from 'ember';
+import { module, test } from 'qunit';
+import Pretender from 'pretender';
+import { authenticateSession } from '../helpers/ember-simple-auth';
+import startApp from '../helpers/start-app';
+
+var App;
+var server;
+
+module('Authentication', {
+  setup: function() {
+    App = startApp();
+    var response = {
+      "errors": [
+        {
+          status: "401",
+          detail: "Access denied"
+        }
+      ]
+    };
+    server = new Pretender(function() {
+      this.get('/api/posts', function() {
+        return [401, { 'Content-Type': 'application/json' }, JSON.stringify(response)];
+      });
+    });
+  },
+  teardown: function() {
+    Ember.run(App, App.destroy);
+    server.shutdown();
+  },
+});
+
+test('user cannot see posts if not authorized', function(assert) {
+  // FIXME: this test fails on any assertion, investigate why
```

In our ember app we do have some global error handling for expected "errors" (eg a 401 Unauthorized).

For example you could add react to all error events on the application route (see [Ember Guides for instructions](https://guides.emberjs.com/v2.3.0/routing/loading-and-error-substates/#toc_the-code-error-code-event). On error status is 401 you could just return and do nothing.

Another possibility is to specify an global `Ember.onerror` function which will be called for all errors. In our application we do this and just ignore errors with status 401 (Redirect to Login Site still happens via `invalidate`).

Hope this helps.